### PR TITLE
Retrieve remote regulating terminal for generators during CGMES import

### LINKED ISSUE #730: Regulating terminal for generators during CGMES import
* **Do you want to request a *feature* or report a *bug*?**
A feature.

* **What is the current behavior?**
When the terminal has not been mapped before importing the regulation control data from a CGMES file, the regulating terminal is null.

* **What is the expected behavior?**
The regulating terminal for generators must be correctly retrieved during the import of CGMES files. 

* **What is the motivation / use case for changing the behavior?**
It makes loadflow after import crash.

* **Please tell us about your environment:**
  - PowSyBl Version: 2.5.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by MioRtia**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines

**Does this PR already have an issue describing the problem ?**
Fixes #730 

**Other information:**
This is inspired from what was done in the CIM1Converter.

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/TerminalMapping.java [hunk 1]
```diff
@@ -39,8 +43,30 @@ public int number(String cgmesTerminalId) {
...
+        if (p.containsKey("TopologicalNode")) {
+            topologicalNodesMapping.computeIfAbsent(p.getId("TopologicalNode"), t -> new ArrayList<>()).add(p.getId("Terminal"));
+        }
+    }
+
+    public boolean areAssociated(String cgmesTerminalId, String topologicalNode) {
+        return topologicalNodesMapping.get(topologicalNode).contains(cgmesTerminalId);
+    }
+
+    public Terminal findFromTopoloficalNode(String topologicalNode) {
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/TerminalMapping.java]: typo, findFromTopolo**g**icalNode

**Comment by zamarrenolm** [comment]
> To avoid getting the `topologicalNode` from the `regulatingControlTerminal` in all the related queries, I think we should only read the terminal and keep the resolution from CGMES terminals to IIDM terminals in the `Context`.
> 
> In fact, I think we should be able to get only a reference to a `RegulatingControl` in the different equipment that may have it (generators, shunts, rtc, ptc) and then build a separate query to gather all information about `RegulatingControl` objects.
> 
> But this should go in a separate change.

I have created issue #737 to handle this.

**Comment by MioRtia** [comment]
> I think we should only read the terminal and keep the resolution from CGMES terminals to IIDM terminals in the `Context`.

I am okay with it in theory but I tested use cases where CGMES terminals are never mapped (which I think is because some TSO use one terminal only for RegulationControl). The only way to find an associated IIDM terminal in this case is to use CGMES topological nodes.

**Comment by zamarrenolm** [comment]
> > I think we should only read the terminal and keep the resolution from CGMES terminals to IIDM terminals in the `Context`.
> 
> I am okay with it in theory but I tested use cases where CGMES terminals are never mapped (which I think is because some TSO use one terminal only for RegulationControl). The only way to find an associated IIDM terminal in this case is to use CGMES topological nodes.

May I take a look at that particular case ?. 

I propose that we merge your current fix to the problem and analyze the other questions in the separate issue I created. Do you agree ?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/TerminalMapping.java (hunk 1 above)
Comment (MioRtia) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/TerminalMapping.java]: Done.

**Comment by MioRtia** [comment]
> > > I think we should only read the terminal and keep the resolution from CGMES terminals to IIDM terminals in the `Context`.
> > 
> > 
> > I am okay with it in theory but I tested use cases where CGMES terminals are never mapped (which I think is because some TSO use one terminal only for RegulationControl). The only way to find an associated IIDM terminal in this case is to use CGMES topological nodes.
> 
> May I take a look at that particular case ?.
> 
> I propose that we merge your current fix to the problem and analyze the other questions in the separate issue I created. Do you agree ?

Yes, I sent you an email related to this issue.
We can do that if this is alright for everyone.

**Pull request merged**